waku/waku_core/topics/sharding.nim

```diff
@@ -91,15 +72,66 @@ proc weightedShardList*(topic: NsContentTopic, shardCount: int, weightList: seq[
 
   list.sort(hashOrder)
 
-  ok(list)
+  ok(list) ]#
 
-proc singleHighestWeigthShard*(topic: NsContentTopic): Result[NsPubsubTopic, string] =
+#[ proc singleHighestWeigthShard*(topic: NsContentTopic): Result[NsPubsubTopic, string] =
   let count = ? shardCount(topic)
 
-  let weights = biasedWeights(count, topic.bias)
+  let weights = repeat(1.0, count)
 
   let list = ? weightedShardList(topic, count, weights)
 
   let (pubsub, _) = list[list.len - 1]
 
-  ok(pubsub)
+  ok(pubsub) ]#
+
+proc genZeroSharding*(topic: NsContentTopic, count: int): NsPubsubTopic =
+  let bytes = toBytes(topic.application) & toBytes(topic.version)
+
+  let hash = sha256.digest(bytes)
+
+  # We only use the last 64 bits of the hash as having more shards is unlikely.
+  let hashValue = uint64.fromBytesBE(hash.data[24..31])
+
+  # This is equilavent to modulo shard count but faster
+  let shard = hashValue and uint64((count - 1))
+ 
+  NsPubsubTopic.staticSharding(ClusterIndex, uint16(shard))
+
+proc autosharding*(topic: NsContentTopic): Result[NsPubsubTopic, string] =
```

Comment:
We _generally_ try to stick to verbs for function names to be clearer that it performs an action. In this case, I'd go for something like `getAutoshard()` or similar. Same applies to `genZeroSharding` above.

Reply:
hmm. `computeShard()` and `computeGen0Shard` then?

Reply:
yeah, works for me!